src/OrleansProviders/Streams/Common/Monitors/IObjectPoolMonitor.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Monitor track block pool related metrics
+    /// </summary>
+    public interface IObjectPoolMonitor
+    {
+        #region Event driven metrics
+        /// <summary>
+        /// Track every time cache allocate memory from block pool
+        /// </summary>
+        /// <param name="blockCount"></param>
+        void TrackObjectAllocatedByCache(int blockCount);
```

Comment:
it is 1.  I don't want to make it void TrackObjectAllocatedByCache(); that's exposing our internal implementation detail: every time object pool allocate, we only allocate one. And less extensible : what if we changed the implementation, and now the object pool can allocate multiple at the same time?